Sources/TuistCache/Utilities/CacheXCFrameworkBuilder.swift

```diff
@@ -51,16 +56,24 @@ public final class CacheXCFrameworkBuilder: CacheArtifactBuilding {
             }
 
             // Build for the device - if required
-            let deviceArchivePath = temporaryDirectory.appending(component: "device.xcarchive")
-            try await self.deviceBuild(
-                projectTarget: projectTarget,
-                scheme: scheme.name,
-                platform: platform,
-                configuration: configuration,
-                archivePath: deviceArchivePath
-            )
-
-            let productNames = deviceArchivePath
+            var deviceArchivePath: AbsolutePath?
+            let buildForDevice = self.cacheOutputType == .xcframework || self.cacheOutputType == .deviceXCFramework
+            if buildForDevice {
+                deviceArchivePath = temporaryDirectory.appending(component: "device.xcarchive")
+                try await self.deviceBuild(
+                    projectTarget: projectTarget,
+                    scheme: scheme.name,
+                    platform: platform,
+                    configuration: configuration,
+                    archivePath: deviceArchivePath!
+                )
+            }
+
+            guard let archivePath = deviceArchivePath ?? simulatorArchivePath else {
+                return
+            }
```

Comment:
Why this was not guarded before?

Reply:
I don't think this can be guarded before, 2 checks must be ran before we can get archivePath. Actually archivePath should always be non-nil, but I don't want to use force cast, so.